pronto_mia/lib/core/services/jwt_token_service.dart

```diff
@@ -38,4 +51,45 @@ class JwtTokenService {
       return _secureStorage.delete(key: _tokenIdentifier);
     }
   }
+
+  Future<bool> hasValidToken() async {
+    final token = await getToken();
+    var tokenValid = false;
+
+    if (token != null && token.isNotEmpty) {
+      try {
+        tokenValid = !JwtDecoder.isExpired(token);
+      } catch (e) {
+        (await _loggingService).log(
+            "JwtTokenService", Level.WARNING, "JWT token could not be decoded");
+        tokenValid = false;

```

Comment:
I don't think it's necessary to set `tokenValid` to false again since it has to be false already when an exception was thrown in the try block.

Reply:
Yes for now this is correct since within the catch clause there is only one assignment. Fixed.